File: src/main/java/io/mfedirko/common/infra/dynamodb/MockContactMeRepository.java
package io.mfedirko.common.infra.dynamodb;

import io.mfedirko.contactme.ContactForm;
import io.mfedirko.contactme.ContactHistory;
import io.mfedirko.contactme.ContactMeRepository;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Repository;

import java.time.Instant;
import java.time.LocalDate;
import java.time.ZonedDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

@Repository
@Profile("mock")
public class MockContactMeRepository implements ContactMeRepository {
    private final List<ContactForm> requests = new ArrayList<>();

    @Override
    public void save(ContactForm form) {
        requests.add(form);
    }

    public List<ContactHistory> findAllContactHistory() {
        return requests.stream().map(form -> ContactHistory.builder()
                        .fullName(form.getFullName())
                        .email(form.getEmail())
                        .messageBody(form.getMessageBody())
                        .build())
                .collect(Collectors.toList());
    }

    @Override
    public List<ContactHistory> findContactHistoryByDate(LocalDate from) {
        return findAllContactHistory();
    }
}

File: src/main/kotlin/io/mfedirko/contactme/MockContactMeRepository.kt
package io.mfedirko.contactme

import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
@Profile("mock")
class MockContactMeRepository : ContactMeRepository {
    private val requests: MutableList<ContactHistory> = ArrayList()
    override fun save(form: ContactForm) {
        requests.add(form.toContactHistory())
    }

    override fun update(vararg contactHistory: ContactHistory) {
        contactHistory.forEach { hist ->
            requests.firstOrNull { it.id == hist.id }
                ?.let {
                    it.fullName = hist.fullName
                    it.email = hist.email
                    it.messageBody = hist.messageBody
                }
                ?: requests.add(hist)
        }
    }

    fun findAllContactHistory(): List<ContactHistory> {
        return requests.toList()
    }

    override fun findContactHistoryByDate(date: LocalDate): List<ContactHistory> {
        return findAllContactHistory()
    }

    override fun findContactHistory(spec: ContactHistorySpec, limit: Int, offset: Int): List<ContactHistory> {
        return findAllContactHistory()
    }

}